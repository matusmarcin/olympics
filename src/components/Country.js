import React, { PropTypes, Component } from 'react';
import Medal from './Medal';

class Country extends Component {
  constructor(props) {
    super(props);
    this.state = { expanded: false };
    this.toggle = this.toggle.bind(this);
  }

  getClassName() {
    return `country 
      ${this.state.expanded ? 'expanded ' : ''}
      ${this.props.goldCount > 0 ? 'has-gold ' : ''}
      ${this.props.silverCount > 0 ? 'has-silver ' : ''}
      ${this.props.bronzeCount > 0 ? 'has-bronze ' : ''}
      `;
  }

  toggle(e = false) {
    if (e) {
      e.preventDefault();
    }
    this.setState({ expanded: !this.state.expanded });
  }

  render() {
    return (
      <li className={this.getClassName()}>
        { // eslint-disable-next-line jsx-a11y/no-static-element-interactions
        }<a onClick={e => this.toggle(e)}>
          <span>
            {this.props.emoji}
            {this.props.name ? this.props.name : this.props.code} ({this.props.totalCount})
          </span>
          <abbr title={`${this.props.goldCount} gold medals`}>{'ü•á'.repeat(this.props.goldCount)}</abbr>
          <abbr title={`${this.props.silverCount} silver medals`}>{'ü•à'.repeat(this.props.silverCount)}</abbr>
          <abbr title={`${this.props.bronzeCount} bronze medals`}>{'ü•â'.repeat(this.props.bronzeCount)}</abbr>
        </a>
        <ul>
          {this.props.medals.map(medal =>
            <Medal
              key={medal.id}
              {...medal}
            />,
          )}
        </ul>
      </li>
    );
  }
}

Country.propTypes = {
  medals: PropTypes.arrayOf(PropTypes.shape({
    athlete: PropTypes.string.isRequired,
    sex: PropTypes.string.isRequired,
    event: PropTypes.string.isRequired,
    medal: PropTypes.string.isRequired,
  }).isRequired).isRequired,
  code: PropTypes.string.isRequired,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  emoji: PropTypes.string,
  goldCount: PropTypes.number.isRequired,
  silverCount: PropTypes.number.isRequired,
  bronzeCount: PropTypes.number.isRequired,
  totalCount: PropTypes.number.isRequired,
};

Country.defaultProps = {
  name: false,
  emoji: 'üè≥',
};

export default Country;
