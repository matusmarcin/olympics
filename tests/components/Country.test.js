/* eslint-env jest, browser */
import React from 'react';
import renderer from 'react-test-renderer';
import Country from '../../src/components/Country';

const countryData = {
  medals: [
    {
      athlete: 'MCLELLAN, Sally',
      country: 'AUS',
      sex: 'Women',
      event: '100m hurdles',
      medal: 'Silver',
      id: 14,
    },
    {
      athlete: 'TALLENT, Jared',
      country: 'AUS',
      sex: 'Men',
      event: '20km walk',
      medal: 'Bronze',
      id: 33,
    },
    {
      athlete: 'TALLENT, Jared',
      country: 'AUS',
      sex: 'Men',
      event: '50km walk',
      medal: 'Silver',
      id: 110,
    },
    {
      athlete: 'HOOKER, Steve',
      country: 'AUS',
      sex: 'Men',
      event: 'pole vault',
      medal: 'Gold',
      id: 160,
    },
  ],
  goldCount: 1,
  silverCount: 2,
  bronzeCount: 1,
  totalCount: 4,
  code: 'AUS',
  name: 'Australia',
  emoji: 'ðŸ‡¦ðŸ‡º',
};

it('renders correctly', () => {
  const tree = renderer.create(
    <Country {...countryData} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('expands on click', () => {
  const tree = renderer.create(
    <Country {...countryData} />,
  ).toJSON();
  tree.children[0].props.onClick();
  expect(tree).toMatchSnapshot();
});
