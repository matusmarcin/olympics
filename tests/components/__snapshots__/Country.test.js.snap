exports[`test expands on click 1`] = `
<li
  className="country 
      
      has-gold 
      has-silver 
      has-bronze 
      ">
  <a
    onClick={[Function]}>
    <span>
      🇦🇺
      Australia
       (
      4
      )
    </span>
    <abbr
      title="1 gold medals">
      🥇
    </abbr>
    <abbr
      title="2 silver medals">
      🥈🥈
    </abbr>
    <abbr
      title="1 bronze medals">
      🥉
    </abbr>
  </a>
  <ul>
    <li>
      🏃‍♀️
      MCLELLAN, Sally
       - 
      100m hurdles
       - 
      <span>
        🥈
      </span>
    </li>
    <li>
      🏃
      TALLENT, Jared
       - 
      20km walk
       - 
      <span>
        🥉
      </span>
    </li>
    <li>
      🏃
      TALLENT, Jared
       - 
      50km walk
       - 
      <span>
        🥈
      </span>
    </li>
    <li>
      🏃
      HOOKER, Steve
       - 
      pole vault
       - 
      <span>
        🥇
      </span>
    </li>
  </ul>
</li>
`;

exports[`test renders correctly 1`] = `
<li
  className="country 
      
      has-gold 
      has-silver 
      has-bronze 
      ">
  <a
    onClick={[Function]}>
    <span>
      🇦🇺
      Australia
       (
      4
      )
    </span>
    <abbr
      title="1 gold medals">
      🥇
    </abbr>
    <abbr
      title="2 silver medals">
      🥈🥈
    </abbr>
    <abbr
      title="1 bronze medals">
      🥉
    </abbr>
  </a>
  <ul>
    <li>
      🏃‍♀️
      MCLELLAN, Sally
       - 
      100m hurdles
       - 
      <span>
        🥈
      </span>
    </li>
    <li>
      🏃
      TALLENT, Jared
       - 
      20km walk
       - 
      <span>
        🥉
      </span>
    </li>
    <li>
      🏃
      TALLENT, Jared
       - 
      50km walk
       - 
      <span>
        🥈
      </span>
    </li>
    <li>
      🏃
      HOOKER, Steve
       - 
      pole vault
       - 
      <span>
        🥇
      </span>
    </li>
  </ul>
</li>
`;
